Products/zms/_builder.py

```diff
@@ -185,18 +185,16 @@ def OnStartElement(self, name, attrs):
             if self.oRootTag == 'ZMS' and 'id' in attrs:
               id = attrs.get( 'id')
               prefix = standard.id_prefix(id)
-              self.getNewId(prefix) # force sequence to generate new-id
-            elif 'id_fix' in attrs:
-              id = attrs.get( 'id_fix')
-              prefix = standard.id_prefix(id)
-              self.getNewId(prefix) # force sequence to generate new-id
+              new_id = self.getNewId(prefix) # force sequence to generate new-id
             elif 'id_prefix' in attrs:
               prefix = attrs.get( 'id_prefix')
               id = self.getNewId(prefix)
             elif 'id' in attrs:

```

Comment:
IMHO the code inside this `elif 'id' in attrs` will never be reached, as the `elif 'id_prefix' in attrs` is always true (at least in my test xml exports is always `id_prefix="e"`) -- so preserving of the e-ids in xml does not work with `id = attrs.get( 'id')`.

As soon as I use `if 'id' in attrs`, the e-ids in xml are taken over as expected if the option "Ignore ZMS-IDs and always create new IDs" is disabled -- and vice versa reassigned if enabled.